### PR TITLE
[FIRRTL] DontTouch on a node should set the symbol on the resulting wire

### PR DESCRIPTION
**Pull request by darthscsi**

### CODE REVIEW

File: lib/Conversion/FIRRTLToHW/LowerToHW.cpp [hunk 1]
```diff
@@ -1609,6 +1609,14 @@ LogicalResult FIRRTLLowering::visitDecl(NodeOp op) {
...
       auto wire = builder.create<sv::WireOp>(operand.getType(), name);
       builder.create<sv::ConnectOp>(wire, operand);
+
+      if (AnnotationSet(op).hasDontTouch()) {
+        auto moduleName = cast<hw::HWModuleOp>(op->getParentOp()).getName();
+        auto symName = builder.getStringAttr(
+            (Twine("__") + moduleName + Twine("__") + name.getValue())
+                .str());
+        wire->setAttr("sym_name", symName);
+      }
```
Comment (seldridge) [inline lib/Conversion/FIRRTLToHW/LowerToHW.cpp:1619]: This looks correct to me. 

Shouldn't this also work for nodes that don't have a name?

**Comment by darthscsi** [comment]
What does it mean to keep a DNT node without a name?  Nothing down stream
can refer to it, since they won't know the name.

**Comment by seldridge** [comment]
I think this would be hard to hit from actual FIRRTL IR as input after: https://github.com/llvm/circt/commit/7f87840027e4e481b291415b427f83db4665a2b6. 

However, something could be unnamed, but still annotated, e.g.:

File: lib/Conversion/FIRRTLToHW/LowerToHW.cpp (hunk 1 above)
Comment (prithayan) [reply lib/Conversion/FIRRTLToHW/LowerToHW.cpp:1619]: Do we need to rely on whether a name is present or not?
Can we just lower it to wires, only if DontTouch annotation is present ?

**Comment by darthscsi** [comment]
> Nothing downstream can refer to it by name, but they can find it via _another_ annotation marking it. 

Are there any annotations which will still exist, not be localized to a node, and not be translated to something non-firrtl specific?

**Review by seldridge — APPROVED** [review]
This is a-okay.

Ignore my comments as they are effectively a change to behavior that may be needed on top of this. :+1:

**Pull request merged**